src/System.IO.FileSystem/tests/Performance/Perf.FileStream.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Perf_FileStream : FileSystemTest
+    {
+        private static byte[] CreateBytesToWrite(int size)
+        {
+            PerfUtils utils = new PerfUtils();
+            return Text.Encoding.UTF8.GetBytes(utils.CreateString(size));
+        }
+
+        private string CreateFile(int size)
+        {
+            PerfUtils utils = new PerfUtils();
+            string filePath = GetTestFilePath();
+            byte[] bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(size));
+            File.WriteAllBytes(filePath, bytes);
+            return filePath;
+        }
+
+        [Benchmark]
+        [InlineData(2000000)]
+        [InlineData(100000000)]
+        public void Write(int size)
+        {
+            string filePath = GetTestFilePath();
+            byte[] bytes = CreateBytesToWrite(size);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (FileStream writer = File.Create(filePath))
+                using (iteration.StartMeasurement())
+                {
+                        writer.Write(bytes, 0, size);
```

Comment:
Good idea. I think I'll rewrite these tests to cover some cases around the optimizations:

- Perform 1 read for full desired size. Skips the buffer entirely.
- Perform x reads each of size _bufferSize - 1 (4095 bytes). Uses the buffer but forces a read every time.
- Perform x reads each of size _bufferSize / 8 (512 bytes). Buffer is useful.

How do those sound?